teps/0044-data-locality-and-pod-overhead-in-pipelines.md

```diff
@@ -301,11 +301,75 @@ controller to make this decision.
+Users will choose which parts of a pipeline to run in a pod by grouping them into a sub-Pipeline, using the [Pipelines in Pipelines](./0056-pipelines-in-pipelines.md) feature.
+Any Pipelines grouped under a Pipeline executed in a pod will also execute in that pod.
+
+We will maintain the 1-1 relationship between Tasks and TaskRuns by creating one TaskRun per Pipeline Task and using it as the source of truth for the status of the
+execution of that Task.
+
+Additional details still under discussion.
+
+### Rollout plan
+
+#### Required Features
+- **Sequential Tasks**. This TEP aims to mitigate the challenges of passing data between subsequent Tasks, such as a `git clone` followed by a `docker build`, so we must
+support sequential Tasks in one pod.
+- **Workspaces and Results**. These features are also critical for passing data between Tasks.
+- **Finally Tasks**. Finally Tasks should be supported for use cases which involve uploading or processing PipelineRun outputs,
+such as data from previous Tasks contained in Workspaces.
+- **Parallel Tasks**. Another primary use case for this TEP is to support parallel Tasks that share data, which today typically must be run in separate pods on the same node.
+Running parallel Tasks on the same pod wouldn't allow computation to be balanced between nodes, but it would allow users to avoid paying performance and UX penalties
+associated with PVCs or other workspace backing options. In addition, `finally` Tasks, which are also a required feature, run in parallel.
+- **Params**. Params are critical to the reusability of many catalog Tasks, and implementation in a pod will require few, if any, changes.
+
+#### Optional Features
+
+The following features are useful for PipelineRuns executed in pods for the same reasons that they are useful in PipelineRuns
+executed in multiple pods. Not all of these features will be equally easy to implement in a pod.
+
+- **`when` expressions**
+- **retries**
+- **timeouts**
+- **Matrixed Tasks (fanned out from params)**
+
+Use cases for the following features are likely impacted by the execution mode: 
+
+- **hermetic fences between Tasks**: see [Hermekton support](#hermekton-support)
+- **Custom Tasks**: Users frequently use Custom Tasks for actions that don’t make sense to run in a pod, so the logic of a Custom Task
+wouldn’t make sense to run in a pod with other Tasks. However, a Pipeline run in a pod could still contain Custom Tasks whose logic is run elsewhere.
+
+The following features don’t yet exist, but we could choose to add them for Pipeline in a pod:
+
+- **Pipeline- or Task-level resource requirements**: Users may be more likely to care about the resource usage of the pod rather than individual containers,
+since Kubernetes uses the pod’s resource requirements to make decisions on scheduling and eviction.
+- **Pipeline-level sidecars**: This feature is requested in [issue 4235](https://github.com/tektoncd/pipeline/issues/4235). 
```

Comment:
maybe i missed it above but probably makes sense to include task level sidecars in the list above

i'm thinking optional but probably not that hard to add?

okay with it being required too - ive been thinking that it would be useful to be able to define a task with just a sidecar that can be easily composed with other tasks (e.g. to provide a docker in docker sidecar) - however that's a bit outside the scope of how this TEP is currently defined

Reply:
good call-- I've added it for now to optional features.